bigbluebutton-client/src/org/bigbluebutton/modules/users/services/MessageReceiver.as

```diff
@@ -698,16 +702,16 @@ package org.bigbluebutton.modules.users.services
     private function handleUserBroadcastCamStartedEvtMsg(msg:Object):void {
       var userId: String = msg.body.userId as String; 
       var streamId: String = msg.body.stream as String;
-      var isHtml5Client: Boolean = msg.body.isHtml5Client as Boolean;
-      
+      var pattern:RegExp = /^([A-z0-9]+)-([A-z0-9]+)-([A-z0-9]+)$/;

```

Comment:
Could you define one variable `pattern` and access it from the two functions rather than keeping 2 copies?

Reply:
You could even extract the whole check `if (pattern.test(streamId)) ` together with the pattern to be a separate helper function

Reply:
Something like this?
```
private function isValidFlashWebcamStream(streamId: String):Boolean{
    var pattern:RegExp = /^([A-z0-9]+)-([A-z0-9]+)-([A-z0-9]+)$/;
    return pattern.test(streamId);
}
```

Reply:
Looks good to me :+1: